packages/node/src/methods/state-channel/create/controller.ts

```diff
@@ -18,62 +19,80 @@ const CREATE_PROXY_AND_SETUP_GAS = 6e6;
  * having been opened via the creation of the multisig.
  * In "creating a channel", this also creates a multisig while sending details
  * about this multisig to the peer with whom the multisig is owned.
+ *
+ * Details: This returns the hash of the multisig deployment transaction so the
+ * caller can async listen to the on chain confirmation of the transaction being
+ * mined. Once the actual multisig is deployed, its _address_ is sent as an event
+ * to the caller via the `NODE_EVENTS.CREATE_CHANNEL` event. This is because
+ * the address of the multisig is not retrievable from the transaction hash
+ * since the multisig is deployed through an internal transaction of a proxy
+ * factory contract.
  * @param params
  */
 export default async function createChannelController(
   requestHandler: RequestHandler,
   params: Node.CreateChannelParams
-): Promise<Node.CreateChannelResult> {
-  const multisigAddress = await deployMinimumViableMultisigAndGetAddress(
+): Promise<Node.CreateChannelTransactionResult> {
+  const {
+    proxyFactory,
+    transactionHash
+  } = await deployMinimumViableMultisigAndGetTransactionHash(
```

Comment:
A slightly nicer API would be `const tx: Transaction = await sendMultisigCreationTxToChain(...)` and then `txHash = tx.hash`.

Reply:
We don't only care about the transaction itself, but also the contract that deploys it so we can listen for the creation and return the multisig address.